src/swap/saga.ts

```diff
@@ -175,14 +185,34 @@ export function* swapSubmitSaga(action: PayloadAction<SwapInfo>) {
 
     const beforeSwapExecutionTimestamp = Date.now()
     quoteToTransactionElapsedTimeInMs = beforeSwapExecutionTimestamp - quoteReceivedAt
-    yield* call(
+    const receipt = yield* call(
```

Comment:
yes indeed - I should have added that I didn't think too hard about when to add the standby transaction since I know this part of the flow is going to change soon

Reply:
I was thinking about that too. I'm not sure that it is correct to add pending tx before it actually sent to the network. If instead of receipt we got an error, how will we handle this?

Reply:
yes what I meant  was to add pending state after submitting to the network, but before receiving the receipt.